@Hang-Lei-NOAA Regarding the MacOS error, I am seeing these messages:

```
/Users/runner/work/NCEPLIBS-g2tmpl/NCEPLIBS-g2tmpl/src/process_4dot2.c:246:9: note: include '<stdio.h>' or provide a declaration of 'fprintf'
/Users/runner/work/NCEPLIBS-g2tmpl/NCEPLIBS-g2tmpl/src/process_4dot2.c:246:9: warning: incompatible implicit declaration of built-in function 'fprintf' [-Wbuiltin-declaration-mismatch]

```
However it appears that `#include <stdio.h>` is already present near the top of process_4dot2.c.

--- a/.github/workflows/MacOS.yml
+++ b/.github/workflows/MacOS.yml
@@ -17,8 +17,9 @@ jobs:
     runs-on: macos-latest
     env:
       FC: gfortran-12
-      CC: gcc-12
-
+      CC: clang   # ✅ use Apple's clang for C
+      CXX: clang++  # ✅ also set C++ compiler if needed
+      
     steps:
 
     - name: checkout

